app/commodities/page.client.tsx

```diff
@@ -9,84 +9,65 @@ import {
 import CommodityFormResponse from '@/components/commodities/CommodityFormResponse';
 import CommodityFormResponseMobile from '@/components/commodities/MobileCommodityFormResponse';
-import layoutConfig from '../_config/layout';
 import GetColor from '@/app/_hooks/colorSelector';
-import {
-  getLocateCommodityFromApi,
-  getLocateCommodityFromMockApi,
-} from './api';
-import { ICommodity, ICommodityFormResponse } from '@/types/';
+import layoutConfig from '../_config/layout';
+import { useGetMockSubmitFormClient } from '../_lib/api-calls';
+import { ICommodity } from '@/types/';
 
 interface IPageClientProps {
-  commodities: ICommodity[] | null;
+  commodities: ICommodity[];
 }
 
 const PageClient: React.FC<IPageClientProps> = ({ commodities }) => {
-  const [isLoading, setIsLoading] = useState(false);
   const [isBuying, setIsBuying] = useState(true);
-  const [submitted, setSubmitted] = useState(false);
-  const [fetchError, setFetchError] = useState<Error | null>(null);
-  const [commodityResponse, setCommodityResponse] = useState<
-    ICommodityFormResponse[]
-  >([]);
+  const [queryString, setQueryString] = useState('');
+  const {
+    data: clientResponse,
+    isLoading,
+    error,
+    mutate,
+  } = useGetMockSubmitFormClient(queryString);
 
   const [isLarge] = useMediaQuery('(min-width: 1024px)');
 
-  const handleSubmit = async (data: SubmitProps) => {
-    const response = await getLocateCommodityFromApi({
-      setIsLoading,
-      setSubmitted,
-      data,
-      setIsBuying,
-    }).finally(() => {
-      setIsLoading(false);
-      setSubmitted(true);
-    });
-
-    if (response instanceof Error) {
-      setFetchError(response as Error);
-      return;
-    }
-    setCommodityResponse(response);
-  };
+  const handleSubmit = async (data: SubmitProps): Promise<void> => {
+    setQueryString('trade/commodity');
+    //   setQueryString(
+    //     `commodityDisplayName=${data.commodityDisplayName.value
+    //       .split(' ')
+    //       .join('%20')}&maxLandingPadSize=${data.maxLandingPadSize}&minSupply=${
+    //       data.minSupply
+    //     }&minDemand=${data.minDemand}&includeFleetCarriers=${
+    //       data.includeFleetCarriers
+    //     }&includeOdyssey=${data.includeOdyssey}&includePlanetary=${
+    //       data.includePlanetary
+    //     }&x=0&y=0&z=0`,
+    //   );
+    if (data.minDemand === 0) setIsBuying(true);
+    if (data.minSupply === 0) setIsBuying(false);
 
-  // TODO: remove this function after development - aslink87
-  const handleExampleSubmit = async () => {
-    const response = await getLocateCommodityFromMockApi({
-      setIsLoading,
-      setSubmitted,
-    }).finally(() => {
-      setIsLoading(false);
-      setSubmitted(true);
-    });
-
-    if (response instanceof Error) {
-      setFetchError(response as Error);
-      return;
-    }
-    setCommodityResponse(response);
+    if (queryString !== '') await mutate();
```

Comment:
@nitishdhar I'd like your opinion here.

We call the swr hook with a queryString(URL). The problem is that the queryString will comprise of data from the form submission. Here I've commented out what that queryString will look like when fully implemented. So the question is what's the best way to pass the dynamic queryString to the swr hook?

Here I opted for setting the queryString as state when the form submits. It works fine but b/c of the async nature I added an if check to make sure the swr hook isn't called until the state is updated. If u remove the if check it still works but the network tab shows a 404 to the default state of the queryString. Is there a better way to pass a string comprised of form data to the swr hook? I guess I just think the if check is ugly but if it works, it works.

Reply:
@aslink87 you are going right about it. Basically you would could use `useState` hooks for all individual form values as we would for normal forms and create the queryString with that -

```
  const [inputValue, setInputValue] = useState('');
  ....

  // Create a query string based on the input values
  const queryString = `/api/v1/${inputValue}.....`;
```

Now in the handleSubmit, you have two options, a) validate each input individually before calling mutate or use some general form validator that validates the entire form data at once using something like zod.